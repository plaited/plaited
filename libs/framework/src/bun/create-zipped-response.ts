export const jsMimeTypes = ['text/javascript;charset=utf-8', 'application/json;charset=utf-8'] as const

export const isJsMimeType = (type: string): type is (typeof jsMimeTypes)[number] =>
  jsMimeTypes.includes(type as (typeof jsMimeTypes)[number])

export const createZippedResponse = (str: string, type: (typeof jsMimeTypes)[number]) => {
  const compressed = Bun.gzipSync(Buffer.from(str))
  return new Response(compressed, {
    headers: {
      'content-type': type,
      'content-encoding': 'gzip',
    },
  })
}
