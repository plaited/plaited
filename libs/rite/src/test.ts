/// <reference lib="dom" />
/// <reference lib="dom.iterable" />

import { TestResultError, TestResult, BrowserSessionResult } from '@web/test-runner-core/browser/session.js'
import { assert, Assertion, AssertionError } from './assert.js'

type TestCallback = (t: Assertion) => Promise<void> | void

interface Test {
  (name: string, cb: TestCallback): void
  skip: (name: string, cb: TestCallback) => void
}

type TestCase = () => Promise<boolean> | boolean

class TestRunner {
  #completedTests = 0
  #failedTests = 0
  #errors: TestResultError[] = []
  #cases: TestCase[] = []
  #timeout = 5_000
  #tests: TestResult[] = []
  updateTimeout(time: number) {
    this.#timeout = time
  }
  get timeout() {
    return this.#timeout
  }
  addCase(test: TestCase) {
    this.#cases.push(test)
  }
  addResult(result: TestResult) {
    this.#tests.push(result)
  }
  addError(obj: TestResultError) {
    this.#errors.push(obj)
  }
  async run(name: string) {
    for (const testCase of this.#cases) {
      try {
        const passed = await testCase()
        this.#completedTests++
        if (!passed) {
          this.#failedTests++
        }
      } catch (error) {
        console.error(error)
      }
    }

    if (this.#completedTests === this.#cases.length) {
      const results: BrowserSessionResult = {
        passed: !this.#failedTests,
        testResults: {
          name,
          suites: [],
          tests: this.#tests,
        },
        errors: this.#errors,
      }
      return results
    } else {
      throw new AssertionError(`Not all test completed`)
    }
  }
}

declare global {
  interface Window {
    __rite_test_runner?: TestRunner
  }
}

window.__rite_test_runner = new TestRunner()

const throwTimeoutError = async () => {
  const timeout = window.__rite_test_runner.timeout
  await assert.wait(timeout)
  throw new Error(`test takes longer than ${timeout}ms to complete`)
}

export const test: Test = (name, callback) => {
  window.__rite_test_runner.addCase(async () => {
    const startTime = performance.now()
    const result: TestResult = {
      name,
      passed: true,
      skipped: false,
    }
    try {
      await Promise.race([callback(assert), throwTimeoutError()])
      const msg = `✓ ${name}`
      // eslint-disable-next-line no-console
      console.log(msg)
    } catch (error) {
      const msg = `✗ ${name}`
      // eslint-disable-next-line no-console
      console.log(msg)
      result.passed = false
      if (error instanceof AssertionError) {
        const obj: Pick<TestResultError, 'actual' | 'expected' | 'message'> = JSON.parse(error.message)
        result.error = { name, stack: `${error.stack}`, ...obj }
        window.__rite_test_runner.addError({ name, ...obj })
      } else {
        throw new Error(error)
      }
    }
    result.duration = performance.now() - startTime
    window.__rite_test_runner.addResult(result)
    return result.passed
  })
}
test.skip = (name: string, _: TestCallback) => {
  window.__rite_test_runner.addCase(() => {
    const result = {
      name,
      passed: true,
      skipped: true,
    }
    window.__rite_test_runner.addResult(result)
    return true
  })
}
