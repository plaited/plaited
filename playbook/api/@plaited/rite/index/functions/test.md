**@plaited/rite** ( [Readme](../../README.md) \| API )

***

[Plaited Typedocs](../../../../modules.md) / [@plaited/rite](../../modules.md) / [index](../README.md) / test

# Function: test()

> **test**(`name`, `cb`): `void`

## Parameters

▪ **name**: `string`

▪ **cb**: `TestCallback`

## Returns

`void`

## Source

[libs/rite/src/test.ts:10](https://github.com/plaited/plaited/blob/0d4801d/libs/rite/src/test.ts#L10)

***

Generated using [typedoc-plugin-markdown](https://www.npmjs.com/package/typedoc-plugin-markdown) and [TypeDoc](https://typedoc.org/)
