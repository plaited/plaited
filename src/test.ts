export * from './test/assert.types.js'
export * from './test/assert.constants.js'
export { assert } from './test/assert.js'
export { findByAttribute } from './test/find-by-attribute.js'
export { findByText } from './test/find-by-text.js'
export { fireEvent } from './test/fire-event.js'
export { match } from './test/match.js'
export { throws } from './test/throws.js'
export { wait } from './utils/wait.js'
export { AssertionError, MissingTestParamsError, TimeoutError } from './test/errors.js'
export { usePlay } from './test/use-play.js'
