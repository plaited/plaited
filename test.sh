#!/bin/bash

# Run yarn
yarn install --immutable

# Run yarn build
yarn build

# Run test
yarn test
