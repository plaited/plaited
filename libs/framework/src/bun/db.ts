import { Database } from "bun:sqlite";

export const db = new Database(":memory:");
db.query(`create table templates;`);