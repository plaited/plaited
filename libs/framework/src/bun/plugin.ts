import type {  BunPlugin } from "bun";
import { db } from "./db.js";

export const plugin: BunPlugin = {
  name: "Plaited loader",
  setup(build) {
    build.onResolve({ filter: /\/route.tsx?$/ }, async ({ path }) => {
      // mark external all non-template files
    })
    build.onLoad({ filter: /\/route.tsx?$/ }, async ({ path }) => {
      // modify the contents of the file to be an export object.
      return {
        contents,
        loader: "tsx",
      };
    });
  },
}