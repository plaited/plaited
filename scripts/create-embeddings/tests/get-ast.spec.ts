import { expect, test, describe } from 'bun:test'
import { getAst } from '../get-typescript-ast'

describe('getAst', () => {
  test('should correctly parse and print TypeScript source code', () => {
    // Arrange
    const path = 'test.ts'
    const source = `
      function hello(name: string): string {
        return \`Hello, \${name}!\`;
      }
    `
    // Act
    const result = getAst(path, source)
    // Assert
    expect(result).toMatchSnapshot()
  })

  test('should handle empty source code', () => {
    // Arrange
    const path = 'empty.ts'
    const source = ''
    // Act
    const result = getAst(path, source)
    // Assert
    expect(result).toMatchSnapshot()
  })

  test('should preserve complex TypeScript syntax', () => {
    // Arrange
    const path = 'complex.ts'
    const source = `
      interface User {
        name: string;
        age: number;
      }

      class UserService {
        private users: User[] = [];

        public addUser(user: User): void {
          this.users.push(user);
        }
      }
    `

    // Act
    const result = getAst(path, source)
    // Assert
    expect(result).toMatchSnapshot()
  })
})
