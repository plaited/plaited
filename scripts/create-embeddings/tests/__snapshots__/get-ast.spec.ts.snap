// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`getAst should correctly parse and print TypeScript source code 1`] = `
{
  "kind": 36,
  "statements": [
    {
      "docs": [],
      "hasDeclareKeyword": false,
      "isAsync": false,
      "isDefaultExport": false,
      "isExported": false,
      "isGenerator": false,
      "kind": 13,
      "name": "hello",
      "overloads": [],
      "parameters": [
        {
          "decorators": [],
          "hasOverrideKeyword": false,
          "hasQuestionToken": false,
          "initializer": undefined,
          "isReadonly": false,
          "isRestParameter": false,
          "kind": 30,
          "name": "name",
          "scope": undefined,
          "type": "string",
        },
      ],
      "returnType": "string",
      "statements": [
        "return \`Hello, \${name}!\`;",
      ],
      "typeParameters": [],
    },
  ],
}
`;

exports[`getAst should handle empty source code 1`] = `
{
  "kind": 36,
  "statements": [],
}
`;

exports[`getAst should preserve complex TypeScript syntax 1`] = `
{
  "kind": 36,
  "statements": [
    {
      "callSignatures": [],
      "constructSignatures": [],
      "docs": [],
      "extends": [],
      "getAccessors": [],
      "hasDeclareKeyword": false,
      "indexSignatures": [],
      "isDefaultExport": false,
      "isExported": false,
      "kind": 19,
      "methods": [],
      "name": "User",
      "properties": [
        {
          "docs": [],
          "hasQuestionToken": false,
          "initializer": undefined,
          "isReadonly": false,
          "kind": 33,
          "name": "name",
          "type": "string",
        },
        {
          "docs": [],
          "hasQuestionToken": false,
          "initializer": undefined,
          "isReadonly": false,
          "kind": 33,
          "name": "age",
          "type": "number",
        },
      ],
      "setAccessors": [],
      "typeParameters": [],
    },
    {
      "ctors": [],
      "decorators": [],
      "docs": [],
      "extends": undefined,
      "getAccessors": [],
      "hasDeclareKeyword": false,
      "implements": [],
      "isAbstract": false,
      "isDefaultExport": false,
      "isExported": false,
      "kind": 2,
      "methods": [
        {
          "decorators": [],
          "docs": [],
          "hasOverrideKeyword": false,
          "hasQuestionToken": false,
          "isAbstract": false,
          "isAsync": false,
          "isGenerator": false,
          "isStatic": false,
          "kind": 26,
          "name": "addUser",
          "overloads": [],
          "parameters": [
            {
              "decorators": [],
              "hasOverrideKeyword": false,
              "hasQuestionToken": false,
              "initializer": undefined,
              "isReadonly": false,
              "isRestParameter": false,
              "kind": 30,
              "name": "user",
              "scope": undefined,
              "type": "User",
            },
          ],
          "returnType": "void",
          "scope": "public",
          "statements": [
            "this.users.push(user);",
          ],
          "typeParameters": [],
        },
      ],
      "name": "UserService",
      "properties": [
        {
          "decorators": [],
          "docs": [],
          "hasAccessorKeyword": false,
          "hasDeclareKeyword": false,
          "hasExclamationToken": false,
          "hasOverrideKeyword": false,
          "hasQuestionToken": false,
          "initializer": "[]",
          "isAbstract": false,
          "isReadonly": false,
          "isStatic": false,
          "kind": 31,
          "name": "users",
          "scope": "private",
          "type": "User[]",
        },
      ],
      "setAccessors": [],
      "staticBlocks": [],
      "typeParameters": [],
    },
  ],
}
`;
